includes/AJAX.php

```diff
@@ -49,6 +51,58 @@ public function __construct() {
 
 		// search a product's attributes for the given term
 		add_action( 'wp_ajax_' . self::ACTION_SEARCH_PRODUCT_ATTRIBUTES, [ $this, 'admin_search_product_attributes' ] );
+
+		// Cancel facebook order by the given order ID
+		add_action( 'wp_ajax_' . self::ACTION_CANCEL_ORDER, [ $this, 'admin_cancel_order' ] );
+	}
+
+
+	/**
+	 * Cancel facebook order by the given order ID
+	 *
+	 * @internal
+	 *
+	 * @since 2.1.0-dev.1
+	 */
+	public function admin_cancel_order() {
+
+		$order = null;
+
+		try {
+
+			if ( ! wp_verify_nonce( Framework\SV_WC_Helper::get_posted_value( 'security' ), self::ACTION_CANCEL_ORDER ) ) {
+				throw new Framework\SV_WC_Plugin_Exception( 'Invalid nonce' );
+			}
+
+			$order_id    = Framework\SV_WC_Helper::get_posted_value( 'order_id' );
+			$reason_code = Framework\SV_WC_Helper::get_posted_value( 'reason_code' );
+
+			if ( empty( $order_id ) ) {
+				throw new Framework\SV_WC_Plugin_Exception( 'Order ID is required' );
+			}
+
+			if ( empty( $reason_code ) ) {
+				throw new Framework\SV_WC_Plugin_Exception( 'Cancel reason is required' );
+			}
+
+			$order = wc_get_order( absint( $order_id ) );
+
+			if ( false === $order ) {
+				throw new Framework\SV_WC_Plugin_Exception( 'A valid Order ID is required' );
+			}
+
+			facebook_for_woocommerce()->get_commerce_handler()->get_orders_handler()->cancel_order( $order, $reason_code );
+
+			wp_send_json_success( 'cancelled' );
+
+		} catch ( Framework\SV_WC_Plugin_Exception $exception ) {
+
+			if ( $order instanceof \WC_Abstract_Order ) {
+				$order->add_order_note( sprintf( __( 'Could not cancel order. Code: "%s", message: "%s"', 'facebook-for-woocommerce' ), $exception->getCode(), $exception->getMessage() ) );
+			}
+
+			wp_send_json_error( $exception->getMessage() );
```

Comment:
the exceptions don't seem having a code, though, and the message is not translatable - if  exceptions messages are meant to be displayed to a user other than a developer they could probably be translated via `__()`

also the `sprintf()` here is missing the `/* translators: Placeholders: */` comment to help translators with the placholders content

Reply:
@unfulvio understood ... the implementation doc suggested to include the error code as well ... but I agree it doesn't look like any of the errors have codes

Reply:
@unfulvio see new commits below, They should address your comments